client/src/search/Typeahead/Typeahead.js

```diff
@@ -169,120 +223,56 @@ export class Typeahead extends React.Component {
-                  .groupBy("config_group_index")
-                  .flatMap((results, group_index) =>
-                    _.map(results, (result, index) => ({
-                      is_first_in_group: index === 0,
-                      group_index,
-                      result,
-                    }))
-                  )
-                  .flatMap(
-                    (
-                      { is_first_in_group, group_index, result },
-                      result_index
-                    ) => {
-                      const adjusted_result_index = needs_pagination_up_control
-                        ? result_index + 1
-                        : result_index;
-
-                      return [
-                        is_first_in_group && (
-                          <li
-                            className="typeahead__header"
-                            key={`group-${group_index}`}
-                          >
-                            {this.config_groups[group_index].group_header}
-                          </li>
-                        ),
-                        <li
-                          key={`result-${adjusted_result_index}`}
-                          className={classNames(
-                            "typeahead__item",
-                            adjusted_result_index === selection_cursor &&
-                              "typeahead__item--active"
+          <AutoSizer>
+            {({ width }) => (
+              <ReactResizeDetector
+                handleWidth
+                onResize={() => {
+                  virtualized_cell_measure_cache.clearAll();
+                }}
+              >
+                {() => (
+                  <AutoHeightVirtualList
+                    className="typeahead__dropdown"
+                    role="listbox"
+                    id={this.menu_id}
+                    aria-expanded={this.show_menu}
+                    height={list_height}
+                    width={width}
+                    list_ref={this.virtualized_list_ref}
+                    scrollToIndex={selection_cursor >= 0 ? selection_cursor : 0}
+                    deferredMeasurementCache={virtualized_cell_measure_cache}
+                    rowHeight={virtualized_cell_measure_cache.rowHeight}
+                    rowCount={_.size(list_items)}
+                    rowRenderer={({
+                      index,
+                      isScrolling,
+                      key,
+                      parent,
+                      style,
+                    }) => (
+                      <CellMeasurer
+                        cache={virtualized_cell_measure_cache}
+                        columnIndex={0}
+                        key={key}
+                        parent={parent}
+                        rowIndex={index}
+                      >
+                        <div style={style}>
+                          {_.isEmpty(matching_results) && (
+                            <li className="typeahead__header">
```

Comment:
For consistency, since everything else is just divs now, should switch to div here too